serving/src/main/java/ai/djl/serving/http/KServeDescribeModelResponse.java

```diff
@@ -0,0 +1,253 @@
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.serving.http;
+
+import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** This class represents the response for KServe model metadata request. */
+public class KServeDescribeModelResponse {
+    private String name;
+    private List<String> versions;
+    private String platform;
+    private List<KServeIO> inputs;
+    private List<KServeIO> outputs;
+
+    /** Constructs a {@code DescribeKServeModelResponse} instance. */
+    public KServeDescribeModelResponse() {
+        versions = new ArrayList<>();
+        inputs = new ArrayList<>();
+        outputs = new ArrayList<>();
+    }
+
+    /**
+     * Sets the model name.
+     *
+     * @param name the model name
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Sets the platform.
+     *
+     * @param platform the platform.
+     */
+    public void setPlatform(String platform) {
+        this.platform = platform;
+    }
+
+    /**
+     * Sets the platform given the engine name.
+     *
+     * @param engineName Name of the engine.
+     */
+    public void setPlatformForEngineName(String engineName) {
+        this.platform = Platform.getKServePlatformForEngine(engineName);
+    }
+
+    /**
+     * Sets the list of a model's versions.
+     *
+     * @param versions versions of a model.
+     */
+    public void setVersions(List<String> versions) {
+        this.versions = versions;
+    }
+
+    /**
+     * Adds the input to the list of inputs.
+     *
+     * @param name name of the input.
+     * @param dataType datatype of the input.
+     * @param shape shape of the input.
+     */
+    public void addInput(String name, DataType dataType, Shape shape) {
+        List<Long> shapeList = Arrays.stream(shape.getShape()).boxed().collect(Collectors.toList());
+        KServeDataType kdt = KServeDataType.getKServeDtForNDArrayDT(dataType);
+        this.inputs.add(new KServeIO(name, kdt.name(), shapeList));
+    }
+
+    /**
+     * Adds the output to the list of outputs.
+     *
+     * @param name name of the output.
+     * @param dataType datatype of the output.
+     * @param shape shape of the output.
+     */
+    public void addOutput(String name, DataType dataType, Shape shape) {
+        List<Long> shapeList = Arrays.stream(shape.getShape()).boxed().collect(Collectors.toList());
+        KServeDataType kdt = KServeDataType.getKServeDtForNDArrayDT(dataType);
+        this.inputs.add(new KServeIO(name, kdt.name(), shapeList));
```

Comment:
I tried these codes and found out that this 'inputs' should be replaced by outputs otherwise there will be a nullPointer error.

Reply:
I think the root problem is that sometimes the  model.describeOutput() will return null not the above one.

Reply:
Oh good catch

Reply:
But the null should be fixed now. I added null check in handler.

Reply:
ok